data-reporting-service/person-service/src/main/java/gov/cdc/etldatapipeline/person/service/PersonService.java

```diff
@@ -9,112 +8,122 @@
-                        .collect(Collectors.toSet()))
-                .peek((key, value) -> log.info("Patient Elastic : {}", value.toString()))
-                .to((key, v, recordContext) -> patientElasticSearchTopicName,
-                        Produced.with(
-                                StreamsSerdes.DataEnvelopeSerde(),
-                                StreamsSerdes.DataEnvelopeSerde()));
-
-
-        // PROVIDER
-        KStream<String, List<ProviderSp>> providerStream = personKStream
-                .filter((k, v) -> v.getCd() != null && v.getCd().equalsIgnoreCase("PRV"))
-                .mapValues(v -> providerRepository.computeProviders(v.getPersonUid()));
-
-        // PROVIDER_REPORTING
-        providerStream
-                .flatMap((k, v) -> v.stream()
-                        .map(p -> KeyValue.pair(
-                                transformer.buildProviderKey(p),
-                                transformer.processData(p, PersonType.PROVIDER_REPORTING)))
-                        .collect(Collectors.toSet()))
-                .peek((key, value) -> log.info("Provider : {}", value.toString()))
-                .to((key, v, recordContext) -> providerReportingOutputTopic,
-                        Produced.with(
-                                StreamsSerdes.DataEnvelopeSerde(),
-                                StreamsSerdes.DataEnvelopeSerde()));
-
-        // PROVIDER_ELASTIC_SEARCH
-        providerStream
-                .flatMap((k, v) -> v.stream()
-                        .map(p -> KeyValue.pair(
-                                transformer.buildProviderKey(p),
-                                transformer.processData(p, PersonType.PROVIDER_ELASTIC_SEARCH)))
-                        .collect(Collectors.toSet()))
-                .peek((key, value) -> log.info("Provider : {}", value.toString()))
-                .to((key, v, recordContext) -> providerElasticSearchOutputTopic,
-                        Produced.with(
-                                StreamsSerdes.DataEnvelopeSerde(),
-                                StreamsSerdes.DataEnvelopeSerde()));
+    public PersonService(PatientRepository patientRepository, ProviderRepository providerRepository, PersonTransformers transformer, KafkaTemplate<String, String> kafkaTemplate) {
+        this.patientRepository = patientRepository;
+        this.providerRepository = providerRepository;
+        this.transformer = transformer;
+        this.kafkaTemplate = kafkaTemplate;
+    }
+
+    @RetryableTopic(
+            attempts = "${spring.kafka.consumer.max-retry}",
+            autoCreateTopics = "false",
+            dltStrategy = DltStrategy.FAIL_ON_ERROR,
+            retryTopicSuffix = "${spring.kafka.dlq.retry-suffix}",
+            dltTopicSuffix = "${spring.kafka.dlq.dlq-suffix}",
+            // retry topic name, such as topic-retry-1, topic-retry-2, etc
+            topicSuffixingStrategy = TopicSuffixingStrategy.SUFFIX_WITH_INDEX_VALUE,
```

Comment:
Any exception during processing except for the listed out exceptions will dump the failed data to the DLT topic right. 
Also will this auto create DLT based on the input topic i.e., nbs-person-retry-1?

Reply:
yes, that is right